File: Android/tuilivekit/voiceroomcore/src/main/java/com/trtc/uikit/livekit/voiceroomcore/SeatGridViewObserver.java
package com.trtc.uikit.livekit.voiceroomcore;

import android.view.View;

import com.tencent.cloud.tuikit.engine.room.TUIRoomDefine;

public abstract class SeatGridViewObserver {
    public void onRoomDismissed(String roomId) {
    }

    public void onKickedOutOfRoom(String roomId, TUIRoomDefine.KickedOutOfRoomReason reason, String message) {
    }

    public void onSeatRequestReceived(VoiceRoomDefine.RequestType type, TUIRoomDefine.UserInfo userInfo) {
    }

    public void onSeatRequestCancelled(VoiceRoomDefine.RequestType type, TUIRoomDefine.UserInfo userInfo) {
    }

    public void onKickedOffSeat(TUIRoomDefine.UserInfo userInfo) {
    }

    public void onUserAudioStateChanged(TUIRoomDefine.UserInfo userInfo, boolean hasAudio,
                                        TUIRoomDefine.ChangeReason reason) {
    }

    public void onSeatViewClicked(View seatView, TUIRoomDefine.SeatInfo seatInfo) {
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/voiceroomcore/SeatGridViewObserver.kt
package com.trtc.uikit.livekit.voiceroomcore

import android.view.View
import com.tencent.cloud.tuikit.engine.room.TUIRoomDefine
import com.trtc.uikit.livekit.voiceroomcore.VoiceRoomDefine.RequestType

abstract class SeatGridViewObserver {
    open fun onRoomDismissed(roomId: String) = Unit
    open fun onKickedOutOfRoom(roomId: String, reason: TUIRoomDefine.KickedOutOfRoomReason, message: String) = Unit
    open fun onSeatRequestReceived(type: RequestType, userInfo: TUIRoomDefine.UserInfo) = Unit
    open fun onSeatRequestCancelled(type: RequestType, userInfo: TUIRoomDefine.UserInfo) = Unit
    open fun onKickedOffSeat(userInfo: TUIRoomDefine.UserInfo) = Unit
    open fun onSeatViewClicked(seatView: View, seatInfo: TUIRoomDefine.SeatInfo) = Unit
}
